plugins/processors/parser/parser.go

```diff
@@ -54,11 +71,12 @@ func (p *Parser) Apply(metrics ...telegraf.Metric) []telegraf.Metric {
 
 		// parse fields
 		for _, key := range p.ParseFields {
+			b64 := slices.Contains(p.Base64Fields, key)
```

Comment:
Hmmm this is `O(N*M/2)` I think... Should we use a map to improve the lookup? Or even better, split that fields into base64 and non-base64 to get `O(1)`...